vyper/ir/optimizer.py

```diff
@@ -689,6 +689,11 @@ def _merge_load(argz, _LOAD, _COPY):
                 initial_dst_offset = dst_offset
                 initial_src_offset = src_offset
                 idx = i
+
+            if not allow_overlap and initial_dst_offset <= initial_src_offset <= dst_offset:
```

Comment:
do you have to do do a + 32 somewhere here?

Reply:
i think the `<=` handles that, see the output of this "partial" overlap:

```vyper
@external
def partial() -> uint256:
    a: uint256 = 1337
    b: uint256 = 0
    c: uint256 = 0
    d: uint256 = 0
    e: uint256 = 0

    c = a
    d = b
    e = c

    return e
```